module-apps/apps-common/popups/lock-popups/PhoneLockedWindow.cpp

```diff
@@ -1,18 +1,24 @@
-// Copyright (c) 2017-2021, Mudita Sp. z.o.o. All rights reserved.
+// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
 // For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
 
 #include "PhoneLockedWindow.hpp"
 #include "PhoneLockedInfoData.hpp"
 
 #include <locks/input/PhoneLockedKeysWhitelist.hpp>
 #include <service-appmgr/Controller.hpp>
+#include <popups/data/PopupData.hpp>
 
 namespace gui
 {
     PhoneLockedWindow::PhoneLockedWindow(app::ApplicationCommon *app, const std::string &name)
         : AppWindow(app, name),
           notificationsModel(std::make_shared<NotificationsModel>(NotificationsListPlacement::LockedScreen))
     {
+        clockWallpaper      = std::make_shared<WallpaperClock>(this, notificationsModel);
+        auto quoteWallpaper = std::make_shared<WallpaperQuote>(app, this);
+        auto logoWallpaper  = std::make_shared<WallpaperLogo>(this);
+        wallpaperPresenter  = std::make_unique<WallpaperPresenter>(app, clockWallpaper, quoteWallpaper, logoWallpaper);
```

Comment:
please pass presenter from application :)

Reply:
Presenter needs wallpapers (safest to force passing them through constructor) and wallpapers need window item to be cnostructed -> thus this forces current order of initialization.

Reply:
`presenter needs wallpapers` tbh - it needs view which it triggers it shall not bother how views are implemented. 
It's just the wrong place to connect the presenter to view. Code here is just an aftermath of lack of: https://github.com/mudita/MuditaOS/pull/3463/files#r781229360